html/modules/custom/ghi_menu/src/Form/MenuLinkDeleteWithSublinksSettingsForm.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Drupal\ghi_menu\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Settings form for the menu links with sublinks deletion module.
+ */
+class MenuLinkDeleteWithSublinksSettingsForm extends ConfigFormBase {
+
+  const CONFIG_NAME = 'menu_links_with_sublinks_deletion.settings';
+
+  /**
+   * Entity Type Manager.
+   */
+  protected EntityTypeManagerInterface $entityTypeManager;
+
+  /**
+   * Constructs the form.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($config_factory);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('entity_type.manager')
+    );
+  }
```

Comment:
This should be simplified to prevent a dependency on the required services of ConfigFormBase. You can use this same example as here: https://github.com/UN-OCHA/ghi-site/blob/d66d3cb4c765378daec564d024fe142c3e2f06cd/html/modules/custom/ghi_base_objects/src/Form/BaseObjectForm.php#L27